packages/gatsby/src/internal-plugins/query-runner/query-compiler.js

```diff
@@ -146,7 +146,12 @@ class Runner {
       return compiledNodes
     }
 
-    const printContext = printTransforms.reduce(
+    // relay-compiler v1.5.0 added "StripUnusedVariablesTransform" to 
+    // printTransforms. Unfortunately it currently doesn't detect variables
+    // in input objects widely used in gatsby, and therefore removing 
+    // variable declaration from queries.
+    // As a temporary workaround remove that transform by slicing printTransforms.
+    const printContext = printTransforms.slice(0, -1).reduce(
```

Comment:
this is very hacky - I pinned relay-compiler to 1.5.0 and have working fix for that locally ( https://github.com/pieh/relay/commit/ccfa12ac3c85a3fca58867ab3633ed693cabea32 ). Will need to get to know with contributing rules on relay monorepo, run tests etc to send PR upstream

Reply:
For reference - https://github.com/facebook/relay/pull/2417